src/main/java/com/github/matidominati/medicalclinic/service/validator/VisitDataValidator.java

```diff
@@ -0,0 +1,63 @@
+package com.github.matidominati.medicalclinic.service.validator;
+
+import com.github.matidominati.medicalclinic.exception.DataAlreadyExistsException;
+import com.github.matidominati.medicalclinic.exception.DataNotFoundException;
+import com.github.matidominati.medicalclinic.exception.IncorrectDateException;
+import com.github.matidominati.medicalclinic.model.dto.commandDto.createCommand.CreateVisitCommand;
+import com.github.matidominati.medicalclinic.model.entity.Doctor;
+import com.github.matidominati.medicalclinic.model.entity.Patient;
+import com.github.matidominati.medicalclinic.model.entity.Visit;
+import com.github.matidominati.medicalclinic.model.enums.VisitType;
+import com.github.matidominati.medicalclinic.repository.VisitRepository;
+import lombok.AllArgsConstructor;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Component;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Optional;
+
+@Component
+@AllArgsConstructor
+public class VisitDataValidator {
```

Comment:
Jak mamy klase ktora ma tylko metody statyczne to raczej nie chcemy tworzyc instancji takiej klasy i nie chcemy zeby byla taka mozliwosc, wiec robimy sobie taka klase
public final class VisitDataValidator  <- final wiec nie mozemy dziedziczyc po niej
i dodajemy jeszcze
@NoArgsConstructor(access = AccessLevel.PRIVATE) <- dzieki czemu bedziemy miec konstruktor prywatny bezargumentwoy a skoro bedzie prywatny to nie bedzie mozna z zewnatrz utworzyc instancji tej klasy